apps/next/src/components/ViewContainerSwitch/ViewContainerSwitch.tsx

```diff
@@ -0,0 +1,48 @@
+import { Box, IconButton } from '@avalabs/k2-alpine';
+import { ContextContainer } from '@core/types';
+import { isSpecificContextContainer, useSettingsContext } from '@core/ui';
+import { FC } from 'react';
+import DockToLeft from './DockToLeft.svg';
+
+const viewInSidePanel = async () => {
+  const currentWindow = await chrome.windows.getCurrent({ populate: true });
+  if (currentWindow?.id) {
+    await chrome.sidePanel.open({
+      windowId: currentWindow.id,
+    });
+  }
+};
+
+const viewInPopup = async () => {
+  await chrome.action.openPopup({});
+};
+
+export const ViewContainerSwitch: FC = () => {
+  const { setPreferredView } = useSettingsContext();
+
+  const onClick = async () => {
+    const isSidePanel = isSpecificContextContainer(ContextContainer.SIDE_PANEL);
+    window.close();
+    if (isSidePanel) {
+      await viewInPopup();
+    } else {
+      await viewInSidePanel();
+    }
+    setPreferredView(isSidePanel ? 'floating' : 'sidebar');
```

Comment:
From my testing, `window.close()` may prevent `setPreferredView()` from sending a request when switching from `floating` to `sidebar`.

The other way around just works, probably because the slide-out animation lets the sidebar live a bit longer.

This (for me) resulted in the `sidebar` preference not being saved:

https://github.com/user-attachments/assets/18716be6-17b9-4917-9a9b-27a6fcf334df

Let's change the order here:


```suggestion
    // Save preferrence before closing the window.
    setPreferredView(isSidePanel ? 'floating' : 'sidebar');

    window.close();

    if (isSidePanel) {
      await viewInPopup();
    } else {
      await viewInSidePanel();
    }
```